x-pack/elastic-agent/pkg/agent/application/monitoring_decorator.go

```diff
@@ -65,8 +65,21 @@ func injectMonitoring(agentInfo *info.AgentInfo, outputGroup string, rootAst *tr
 		monitoringOutputName = monitoringOutputNameKey.String()
 	}
 
+	typeValue := elasticsearchKey
+	t, found := transpiler.Lookup(rootAst, fmt.Sprintf("%s.%s.type", outputsKey, monitoringOutputName))
+	if found {
+		v, ok := t.Value().(*transpiler.StrVal)
+		if !ok {
+			return programsToRun, nil
+		}
+		typeValue = v.String()
+	} else {
+		typeValue = elasticsearchKey
+
+	}
```

Comment:
nit; It seems we have this pattern in more than one place. How about introducing a `transpiler.LookupString(ast, key) (contents string, ok bool)`?